pkg/apis/core/v1/conversion.go

```diff
@@ -270,6 +272,98 @@ func Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec(in *v1.PodTemplateSpec,
 	return nil
 }
 
+// helper for v1.PodIPInfo list to core.PodIPinfoList
+func v1_PodIPs_To_core_PodIPs(inList []v1.PodIPInfo) []core.PodIPInfo {
+	out := make([]core.PodIPInfo, len(inList))
+	for idx, value := range inList {
+		out[idx] = core.PodIPInfo{
+			IP:         value.IP,
+			Properties: value.Properties,
+		}
+	}
+	return out
+}
+
+// Converts podStatus from v1 to core
+// auto + handling podIp/podIps
+func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatus, s conversion.Scope) error {
+	if err := autoConvert_v1_PodStatus_To_core_PodStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// handle ip/ips fields
+	if 0 < len(in.PodIP) && 0 < len(in.PodIPs) {
+		// both fields are provided
+		if in.PodIP != in.PodIPs[0].IP {
+			return fmt.Errorf("conversion Error: in.PodIP(%v) != in.PodIPs[0](%v)", in.PodIP, in.PodIPs[0].IP)
+		}
+		out.PodIPs = v1_PodIPs_To_core_PodIPs(in.PodIPs)
```

Comment:
My assumption (been a while since I took this apart) was that the autoConvert above would do the "obvious" conversion, which is what the helper function does.  Unless I am missing something, I would guess you don't need this at all

Reply:
083bf4e moves away from custom conversions and depends completely on `autoConvert_*` family of functions, thanks for the tip.